### ISSUE TITLE
Add yearly reading goals to mobile My Books page

### ISSUE THREAD

**Issue by jimchamp** [internetarchive/openlibrary#10168]
### Proposal

At some point, the yearly reading goals affordances were removed from the [My Books page](https://openlibrary.org/account/books) for small screen sizes.  As a result, patrons can no longer set a yearly reading goal when browsing the site on their phones.

The reading goal affordances were recently moved to the sidebar for desktop views (related [PR](https://github.com/internetarchive/openlibrary/pull/10085)).  Mobile views do not display the sidebar, so the reading goal affordances will need to be placed somewhere else on the page.  Adding them between the "Privacy Settings" chip and the "My Loans" link seems like a good place for now.

![Screenshot 2024-12-18 134131](https://github.com/user-attachments/assets/df37b789-9e63-46fa-bc74-cf76fde6c8fc)

### Justification

A solution to this issue will allow our patrons to set reading goals from their phones.

#### Requirements Checklist

* [ ] Yearly reading goal CTA link and progress bar are present on the logged-in patron's My Books page in mobile views
* [ ] Duplicate markup for the yearly reading goal affordances, if any, are moved to a new template in [`openlibrary/templates/check_ins`](https://github.com/internetarchive/openlibrary/tree/master/openlibrary/templates/check_ins)

#### Related files

* Existing templates for yearly reading goals can be found [here]([`openlibrary/templates/check_ins`](https://github.com/internetarchive/openlibrary/tree/master/openlibrary/templates/check_ins))
* Markup for yearly reading goal affordances that are found in the sidebar can be found [here](https://github.com/internetarchive/openlibrary/blob/e010b2a13697de70170033902ba2e27a1e1acbe9/openlibrary/templates/account/sidebar.html#L23-L44)

#### Stakeholders

*

<hr>

#### Instructions for Contributors

Please [run these commands](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#working-on-your-branch) to ensure your repository is up to date **before** [creating a new branch](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#making-changes-and-creating-a-pull-request) to work on this issue and **each time after** pushing code to Github, because the pre-commit bot may add commits to your PRs upstream.

**Comment by sumana-2705** [comment]
@jimchamp  can I work on this issue?

**Comment by jimchamp** [comment]
@sumana-2705, of course!  Let me know if you have any questions about anything.

**Comment by mekarpeles** [comment]
@sumana-2705 as a first step, feel free to share any designs you think may be effective for implementing this and we can offer you feedback.

Here's how it currently looks on desktop:
<img width="476" alt="Screenshot 2024-12-23 at 11 34 14 AM" src="https://github.com/user-attachments/assets/7f469eea-1a47-49c7-a36a-6706679a6f14" />

Perhaps something like this (keep the footprint as small as possible) that also says the year, like "YYYY reading goal:"
<img width="444" alt="Screenshot 2024-12-23 at 11 42 44 AM" src="https://github.com/user-attachments/assets/bab7a9bd-94a9-4d73-9c07-9f0225e4a5ac" />

**Comment by mekarpeles** [comment]
Better yet, it may be easier, more consistent, a better experience, and a better use of space if we did something like this (instead of adding more clutter):

<img width="427" alt="Screenshot 2024-12-23 at 11 47 03 AM" src="https://github.com/user-attachments/assets/361331d6-b9f5-44a2-a151-2b262972e067" />

**Comment by sumana-2705** [comment]
Yeah @mekarpeles I was also thinking about the same idea as you have suggested. Thank you for your support. It has taken a few days for me to understand and set up the website in my local machine. I'll soon open a PR as you have suggested.

**Comment by mekarpeles** [comment]
@sumana-2705 if you have any questions, please let me know :)
